package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,537 @@
+                \-- [value] <int>, gives integration step
+
+    .. note::
+        The reader does not currently read mass or charge data.
+
+    .. versionadded:: 2.0.0
+
+    """
+
+    format = 'H5MD'
+    # units is defined as instance-level variable and set from the
+    # H5MD file in __init__() below
+
+    # This dictionary is used to translate H5MD units to MDAnalysis units.
+    # (https://nongnu.org/h5md/modules/units.html)
+    _unit_translation = {
+        'time': {
+            'ps': 'ps',
+            'fs': 'fs',
+            'ns': 'ns',
+            'second': 's',
+            'sec': 's',
+            's': 's',
+            'AKMA': 'AKMA',
+            },
+        'length': {
+            'Angstrom': 'Angstrom',
+            'angstrom': 'Angstrom',
+            'A': 'Angstrom',
+            'nm': 'nm',
+            'pm': 'pm',
+            'fm': 'fm',
+            },
+        'velocity': {
+            'Angstrom ps-1': 'Angstrom/ps',
+            'A ps-1': 'Angstrom/ps',
+            'Angstrom fs-1': 'Angstrom/fs',
+            'A fs-1': 'Angstrom/fs',
+            'Angstrom AKMA-1': 'Angstrom/AKMA',
+            'A AKMA-1': 'Angstrom/AKMA',
+            'nm ps-1': 'nm/ps',
+            'nm ns-1': 'nm/ns',
+            'pm ps-1': 'pm/ps',
+            'm s-1': 'm/s'
+            },
+        'force':  {
+            'kJ mol-1 Angstrom-1': 'kJ/(mol*Angstrom)',
+            'kJ mol-1 nm-1': 'kJ/(mol*nm)',
+            'Newton': 'Newton',
+            'N': 'N',
+            'J m-1': 'J/m',
+            'kcal mol-1 Angstrom-1': 'kcal/(mol*Angstrom)',
+            'kcal mol-1 A-1': 'kcal/(mol*Angstrom)'
+            }
+    }
+    _Timestep = Timestep
+    _data_keywords = ('step', 'lambda')
+
+    def __init__(self, filename, convert_units=True, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str or :class:`h5py.File`
+            trajectory filename or open h5py file
+        convert_units : bool (optional)
+            convert units to MDAnalysis units
+        **kwargs : dict
+            General reader arguments.
+
+        Raises
+        ------
+        RuntimeError
+            when `H5PY`_ is not installed
+        RuntimeError
+            when a unit is not recognized by MDAnalysis
+        ValueError
+            when ``n_atoms`` changes values between timesteps
+        ValueError
+            when ``convert_units=True`` but the H5MD file contains no units
+        NoDataError
+            when the H5MD file has no 'position', 'velocity', or
+            'force' group
+
+        """
+        if not HAS_H5PY:
+            raise RuntimeError("Please install h5py")
+        super(H5MDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.convert_units = convert_units
+        self.driver = kwargs.get('driver', None)
+        self.comm = kwargs.get('comm', None)
+        self.parallel = self.driver and self.comm
```

Comment:
Is it true that you need both? Eg does `driver` default to something sensible if a `comm` is supplied?

We can leave it as it is at the moment but possibly add a comment.